langgraph-sls-fastapi-rag/src/rag_graph/retrieval.py

```
@@ -0,0 +1,135 @@
+
+
+@contextmanager
+def make_pinecone_retriever(
+    configuration: IndexConfiguration, embedding_model: Embeddings
+) -> Generator[VectorStoreRetriever, None, None]:
+    """Configure this agent to connect to a specific pinecone index."""
+    from langchain_pinecone import PineconeVectorStore
+
+    search_kwargs = configuration.search_kwargs
+
+    search_filter = search_kwargs.setdefault("filter", {})
```

Ruff (0.8.2)

Local variable `search_filter` is assigned to but never used

Remove assignment to unused variable `search_filter`

(F841)